pkg/errors/wrapped.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+type wrappedError struct {
```

Comment:
what's the relationship with immuError under pkg/errors? I'd expect immuError to be enough but there may be something I'm not taking into account

Reply:
wrappedError is required to implement wrapping capability and it contains the "cause" property.
immuError could be the cause of a wrappedError or can be returned directly.